### PR TITLE
Fixing the SPN authentication issue and upgrading all azure libraries…

### PR DESCRIPTION
**Pull request by aravindarc**
… to recent version(as of this commit), with versions mentioned in requirements.txt for proper tracking.

This was tested manually using chaos tool also, the SPN issue was resolved and also the experiments are working as expected.

#129 is related to this issue.

The problem was requirements.txt had no versions, because of which latest libraries are being used, which in turn contains breaking changes in apis.

The solution: I upgraded all libraries to use recent version and made the changes required. Following this documentation https://docs.microsoft.com/en-us/azure/developer/python/azure-sdk-overview#migrating-from-older-management-libraries

I have also marked the versions in the requirements.txt to prevent any future version related issue.

Signed-off-by: Aravinda Kumar <aravindrckumar@gmail.com>

### CODE REVIEW

**Comment by aravindarc** [comment]
@Lawouach @bugra-derre Can you please verify this PR and merge it. This is a show stopper for me and the issue is existing for a while

**Comment by Lawouach** [comment]
Looking at this now :)

File: chaosazure/common/resources/graph.py [hunk 1]
```diff
@@ -9,6 +9,7 @@
 
 from chaosazure import init_resource_graph_client
 from chaosazure.common.config import load_configuration
+import azure.mgmt.resourcegraph as arg
```
Comment (Lawouach) [inline chaosazure/common/resources/graph.py:12]: Imports should follow an order such as:

```
standard lib imports

third party imports

package imports
```

Here, this import should be on line 8.

File: chaosazure/common/resources/graph.py [hunk 2]
```diff
@@ -29,15 +30,17 @@ def fetch_resources(input_query: str, resource_type: str,
         raise InterruptExecution(msg)
 
     # prepare results
-    results = __to_dicts(resources.data, client.api_version)
+    results = __to_dicts(resources.data, "2021-03-01")
```
Comment (Lawouach) [inline chaosazure/common/resources/graph.py:33]: What's the impact in future release of pinning this?

File: chaosazure/common/resources/graph.py [hunk 3]
```diff
@@ -29,15 +30,17 @@ def fetch_resources(input_query: str, resource_type: str,
...
     return results
 
 
 def __query_request_from(query, experiment_configuration: Configuration):
     configuration = load_configuration(experiment_configuration)
+    argQueryOptions = arg.models.QueryRequestOptions(result_format="table")
```
Comment (Lawouach) [inline chaosazure/common/resources/graph.py:39]: we follow pep8 naming conventions, here lower_case.

File: requirements.txt [hunk 4]
```diff
@@ -1,8 +1,9 @@
 chaostoolkit-lib>==1.1.2
-azure-mgmt-compute
-azure-mgmt-resourcegraph
-azure-mgmt-web
+azure-mgmt-compute===20.0.0
+azure-mgmt-resourcegraph===8.0.0
+azure-mgmt-web===3.0.0
+azure-identity===1.6.0
```
Comment (Lawouach) [inline requirements.txt:5]: strict pinning is a pain, can we at least be more flexible with something like:

```
azure-mgmt-compute~=20.0
azure-mgmt-resourcegraph~=8.0
azure-mgmt-web~=3.0
azure-identity~=1.6
```

That will allow minor versions only.

File: requirements.txt [hunk 5]
```diff
@@ -1,8 +1,9 @@
...
 dateparser
 logzero
 requests
-msrestazure
\ No newline at end of file
+msrestazure===0.6.4
```
Comment (Lawouach) [inline requirements.txt:9]: likewise.

File: setup.py [hunk 6]
```diff
@@ -7,11 +7,11 @@
 import setuptools
 
 sys.path.insert(0, ".")
-from chaosazure import __version__
 sys.path.remove(".")
 
 name = 'chaostoolkit-azure'
 desc = 'Chaos Toolkit Extension for Microsoft Azure'
+version = '0.9.0'
```
Comment (Lawouach) [inline setup.py:14]: why this change?

**Review by Lawouach — CHANGES_REQUESTED** [review]
Thanks for the hardwork.

Many comments that need to be addressed before we can make a decision here I'm afraid.

Also, could you add a changelog entry, indicating if there is a breaking change as well. Cheers.

File: chaosazure/common/resources/graph.py (hunk 2 above)
Comment (aravindarc) [reply chaosazure/common/resources/graph.py:33]: The **ResourceGraphClient** class doesnot have an **api_version** attribute, the api version is an attribute in the protected member attribute _config, you have to access it like client._config.api_version. Suggest what should be done.

I suggest we remove the version variable from __to_dicts function signature, and stop doing a if else inside, support only >= 2019-04-01 version, since we are going to pin only new versions in the requirements.

File: setup.py (hunk 6 above)
Comment (aravindarc) [reply setup.py:14]: when I try to install chaostoolkit-azure as a standalone dependency without wheel packages or cache, 
like `pip install chaostoolkit-azure --no-cache-dir --no-binary :all:` import is failing with the following error:

```
Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-n1z_3n35/chaostoolkit-azure/setup.py", line 10, in <module>
        from chaosazure import __version__
      File "./chaosazure/__init__.py", line 7, in <module>
        from azure.mgmt.compute import ComputeManagementClient
    ModuleNotFoundError: No module named 'azure'
```

 importing `__version__` in setup.py causes this failure, because by importing this variable, the import statements in chaosazure/__init__.py are executed before requirements are resolved down below in the setup.py

So I think the version has to be hardcoded in the setup.py

**Comment by aravindarc** [comment]
@Lawouach About the changelog entry, there is already an entry in the top marked as unreleased, what should be done about this. 
Also should I increase the version, if yes what should be the version, or will this be done by you.

File: setup.py (hunk 6 above)
Comment (Lawouach) [reply setup.py:14]: I see. Another way would be a utility function such as:

https://github.com/chaostoolkit/chaostoolkit-kubernetes/blob/master/setup.py#L10

then

https://github.com/chaostoolkit/chaostoolkit-kubernetes/blob/master/setup.py#L64

**Comment by Lawouach** [comment]
Add new entries under the "unreleased" section under Changed/Added. Do not bump the version. I'll do that when I release after merging.

**Review by Lawouach — APPROVED** [review]
Fantastic!
